bson/src/main/org/bson/codecs/pojo/PojoCodecProvider.java

```diff
@@ -93,6 +93,13 @@ private <T> PojoCodec<T> getPojoCodec(final Class<T> clazz, final CodecRegistry
         return null;
     }
 
+    private static <T> PojoCodec<T> createCodec(final ClassModel<T> classModel, final CodecRegistry codecRegistry,
+            final List<PropertyCodecProvider> propertyCodecProviders, final DiscriminatorLookup discriminatorLookup) {
+        return shouldSpecialize(classModel)
+                ? new PojoCodecImpl<>(classModel, codecRegistry, propertyCodecProviders, discriminatorLookup)
+                : new LazyPropertyModelCodec.NeedSpecializationCodec<>(classModel, discriminatorLookup);
```

Comment:
`LazyPropertyModelCodec.createCodec` is the place that replaces unusable `NeedSpecializationCodec` with a working `PojoCodecImpl`, which is why `LazyPropertyModelCodec` looks like the best place for this class.

Ideally, `NeedSpecializationCodec` should not exist, and instead of creating it, a working codec should be created right away. Maybe we'll be able to refactor it out, but not in this PR.